compiler/vm/src/heap/mod.rs

```diff
@@ -307,8 +308,8 @@ impl Heap {
     pub fn create_text(&mut self, text: String) -> Pointer {
         self.create(Data::Text(Text { value: text }))
     }
-    pub fn create_symbol(&mut self, symbol: String) -> Pointer {
-        self.create(Data::Symbol(Symbol { value: symbol }))
+    pub fn create_tag(&mut self, symbol: String, value: Option<Pointer>) -> Pointer {
```

Comment:
By slightly modifying the argument type, this function can also be called with a `Pointer` directly since any `T` `impl`s `Into<Option<T>>`:

```suggestion
    pub fn create_tag(&mut self, symbol: String, value: impl Into<Option<Pointer>>) -> Pointer {
```